python/box.tests/test/org/jetbrains/kotlin/python/test/BasicIrBoxTest.kt

```diff
@@ -138,23 +142,37 @@ abstract class BasicIrBoxTest(
             }
 
             if (!skipRegularMode) {
-                val compiledModule = compile(
-                    project = config.project,
-                    mainModule = MainModule.SourceFiles(filesToCompile),
-                    analyzer = AnalyzerWithCompilerReport(config.configuration),
-                    configuration = config.configuration,
-                    phaseConfig = phaseConfig,
-                    irFactory = IrFactoryImpl,
-                    allDependencies = resolvedLibraries,
-                    friendDependencies = emptyList(),
-                    mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
-                    exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
-                    generateFullJs = true,
-                    generateDceJs = runIrDce,
-                    es6mode = runEs6Mode,
-                    multiModule = splitPerModule || perModule,
-                    propertyLazyInitialization = propertyLazyInitialization,
-                )
+                var compilationException: KotlinCompilationException? = null
+                val compiledModuleWithDuration: Pair<Long, CompilerResult?> = measureTimeMillisWithResult {
+                    try {
+                        compile(
+                            project = config.project,
+                            mainModule = MainModule.SourceFiles(filesToCompile),
+                            analyzer = AnalyzerWithCompilerReport(config.configuration),
+                            configuration = config.configuration,
+                            phaseConfig = phaseConfig,
+                            irFactory = IrFactoryImpl,
+                            allDependencies = resolvedLibraries,
+                            friendDependencies = emptyList(),
+                            mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
+                            exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
+                            generateFullJs = true,
+                            generateDceJs = runIrDce,
+                            es6mode = runEs6Mode,
+                            multiModule = splitPerModule || perModule,
+                            propertyLazyInitialization = propertyLazyInitialization,
+                        )
+                    } catch (e: Throwable) {
+                        compilationException = KotlinCompilationException("Message should be set later. If you see this, it's a bug!", e)
```

Comment:
If we don't need the message, can we just make `compilationException` `Throwable` and do `compilationException = e` here (and remove `.cause` below)?

Reply:
Yeah, of course :facepalm: I left this mess after experimenting with some approaches. Thanks, will simplify!

Reply:
Fixed in https://github.com/krzema12/kotlin-python/pull/23/commits/55c1e0c01f0dab3315f90d9c2450c9df0bd21661